app/model/OnlineTestCommands.scala

```diff
@@ -25,42 +25,46 @@ import reactivemongo.bson.{BSONDocument, BSONHandler, Macros}
 
 object OnlineTestCommands {
   case class Phase1Test(scheduleId: Int,
-                        usedForResults: Boolean,
-                        cubiksUserId: Int,
-                        testProvider: String = "cubiks",
-                        token: String,
-                        testUrl: String,
-                        invitationDate: DateTime,
-                        participantScheduleId: Int,
-                        startedDateTime: Option[DateTime] = None,
-                        completedDateTime: Option[DateTime] = None,
-                        resultsReadyToDownload: Boolean = false,
-                        reportId: Option[Int] = None,
-                        reportLinkURL: Option[String] = None,
-                        reportStatus: Option[String] = None)
+    usedForResults: Boolean,
+    cubiksUserId: Int,
+    testProvider: String = "cubiks",
+    token: String,
+    testUrl: String,
+    invitationDate: DateTime,
+    participantScheduleId: Int,
+    startedDateTime: Option[DateTime] = None,
+    completedDateTime: Option[DateTime] = None,
+    resultsReadyToDownload: Boolean = false,
+    reportId: Option[Int] = None,
+    reportLinkURL: Option[String] = None,
+    reportStatus: Option[String] = None,
+    testResult: Option[model.persisted.TestResult] = None
+  )
 
   object Phase1Test {
     import repositories.BSONDateTimeHandler
     implicit val phase1TestHandler: BSONHandler[BSONDocument, Phase1Test] = Macros.handler[Phase1Test]
     implicit val phase1TestFormat = Json.format[Phase1Test]
   }
 
-  case class Phase1TestProfile(
-                                expirationDate: DateTime,
-                                tests: List[Phase1Test]
-  ) {
-    def activeTests = tests filter (_.usedForResults)
+  trait TestProfile {
+    def expirationDate: DateTime
+    def tests: List[Phase1Test]
+  }
 
-    def hasNotStartedYet = activeTests.forall(_.startedDateTime.isEmpty)
+  case class Phase1TestProfile(expirationDate: DateTime,
+    tests: List[Phase1Test]
+  ) extends TestProfile {
 
+    def activeTests = tests filter (_.usedForResults)
+    def hasNotStartedYet = activeTests.forall(_.startedDateTime.isEmpty)
     def hasNotCompletedYet =  activeTests.exists(_.completedDateTime.isEmpty)
-
     def hasNotResultReadyToDownloadForAllTestsYet =  activeTests.exists(!_.resultsReadyToDownload)
   }
 
   object Phase1TestProfile {
     import repositories.BSONDateTimeHandler
-    implicit val phase1TestProfileHandler: BSONHandler[BSONDocument, Phase1TestProfile] = Macros.handler[Phase1TestProfile]
+    implicit def bsonHandler: BSONHandler[BSONDocument, Phase1TestProfile] = Macros.handler[Phase1TestProfile]
```

Comment:
Not sure, I'll change it back.  Think it was when I was playing around with the function types in the OnlineTestRepo trait.